### ISSUE TITLE
RKE2 clusters - Snapshots are not available on the UI for a cluster with cloud provider enabled

### ISSUE THREAD

**Issue by sowmyav27** [rancher/dashboard#5476]
**on 2.6-head commit id: `c2d8e32`**

- Deploy an RKE2 cluster - AWS ec2 node driver with Cloud provider enabled on the cluster
- take snapshots on the cluster
- Navigate to Cluster Management --> cluster details --> Snapshots tab
- No snapshots are listed
- Snapshots are available when I ssh into the etcd node:
```
/var/lib/rancher/rke2/server/db/snapshots# ls -ltr
total 38268
-rw------- 1 root root 13058080 Mar 23 05:00 etcd-snapshot-<etcd node name>-1648011600
-rw------- 1 root root 13058080 Mar 23 05:06 on-demand-sowmya-cp-pool1-59523e22-blrvg-1648011980
-rw------- 1 root root 13058080 Mar 23 05:32 on-demand-sowmya-cp-pool1-59523e22-blrvg-1648013523
```

**Note:**
- On a DO Node driver cluster, the snapshots are available on the UI. 
- On an EC2 node driver with no Cloud provider enabled, snapshots are available on the UI.

**Comment by catherineluse** [comment]
I thought I would have time to work on this but I didn't. Maybe someone else can pick it up?

**Comment by rak-phillip** [comment]
I can pick this up

**Comment by rak-phillip** [comment]
I created 4 snapshots total:

* 2 for an RKE2 cluster on EC2 WITH Cloud Provider
* 2 for an RKE2 cluster on EC2, NO Cloud Provider

I only ever see 2 snapshots total when we get a collection of all our snapshots, these snapshots are for the NO Cloud Provider cluster. I was initially thinking that we were filtering results incorrectly, but we can't filter these snapshots out if we don't ever receive them. 

I may need someone from the backend team to provide feedback as to why we might not be returning all of the snapshots for WITH Cloud Provider.

@gaktive @sowmyav27

**Comment by gaktive** [comment]
Thanks @rak-phillip -- is there a related backend ticket? or did you talk to someone already for help?

**Comment by rak-phillip** [comment]
@gaktive there's no backend ticket that I'm aware of..

**Comment by rak-phillip** [comment]
@gaktive here's our new ticket rancher/rancher#37052

**Comment by gaktive** [comment]
@rak-phillip there's a question for you about possible logs in https://github.com/rancher/rancher/issues/37052#issuecomment-1088725284

**Comment by rak-phillip** [comment]
@gaktive thanks for the heads up! I sent @Oats87 a DM with a setup that has this reproduced. Hopefully, we'll be able to collect more details.

**Comment by gaktive** [comment]
This is unblocked now per https://github.com/rancher/rancher/issues/37052

**Comment by rak-phillip** [comment]
@gaktive this works as expected, no UI change needed

**Comment by slickwarren** [comment]
tested on v2.6-head (6c12bee):
using an rke2 cluster with (vsphere) cloud provider enabled, the UI is showing both manual and recurring snapshots
